api/python/quilt3/telemetry.py

```diff
@@ -0,0 +1,146 @@
+import atexit
+from concurrent.futures import ThreadPoolExecutor, wait
+import functools
+import os
+import platform
+from requests_futures.sessions import FuturesSession
+import sys
+from threading import Lock
+import uuid
+
+from .util import load_config, set_config_value
+from . import __version__ as quilt3_version
+
+TELEMETRY_URL = "https://telemetry.quiltdata.cloud/Prod/metrics"
+TELEMETRY_USER_AGENT = "QuiltCli"
+TELEMETRY_CLIENT_TYPE = "quilt3-python-client"
+TELEMETRY_SCHEMA_VERSION = "pyclient-usage-metrics-v0"
+
+DISABLE_USAGE_METRICS_ENVVAR = "QUILT_DISABLE_USAGE_METRICS"
+MAX_CLEANUP_WAIT_SECS = 5
+
+
+
+
+class ApiTelemetry:
+    session = None
+    session_id = str(uuid.uuid4())
+    pending_reqs = []
+    pending_reqs_lock = Lock()
+    telemetry_disabled = False
+
+    @staticmethod
+    def create_session():
+        return FuturesSession(executor=ThreadPoolExecutor(max_workers=2))
+
+    def __init__(self, api_name):
+        ApiTelemetry.telemetry_disabled = ApiTelemetry.telemetry_is_disabled()
+
+        if ApiTelemetry.session is None:
+            ApiTelemetry.session = ApiTelemetry.create_session()
+
+        self.api_name = api_name
+
+    @staticmethod
+    def has_connectivity():
+        # TODO: Implement this check. Punting for the near term
+        return True
+
+    @staticmethod
+    def check_telemetry_disabled_in_config():
+        """
+        Check if 'telemetry_disabled' field exists in quilt3 config. If it does, return it. If it does not exist, set
+        it to default value of 'false' (to handle case of current users who predate this config field).
+        """
+        config_value = load_config().get("telemetry_disabled")
+        if config_value is not None:
+            return config_value
+        else:
+            set_config_value("telemetry_disabled", False)
+            return False
+
+
+    @staticmethod
+    def check_telemetry_disabled_by_envvar():
+        # TODO: Potential bug where 'QUILT_DISABLE_USAGE_METRICS=False' doesn't work as expected
+        return bool(os.environ.get(DISABLE_USAGE_METRICS_ENVVAR, False))
```

Comment:
might be slightly more correct to default to `''` instead of `False` because when it's set `environ.get` will give you a string.